pkg/controller/kogitoapp/kogitoapp_controller.go

```diff
@@ -162,32 +164,75 @@ func (r *ReconcileKogitoApp) Reconcile(request reconcile.Request) (reconcile.Res
 		},
 	})
 	if err != nil {
-		return reconcile.Result{}, err
+		updateResourceResult.Err = err
+		updateResourceResult.ErrorReason = v1alpha1.ParseCRRequestFailedReason
 	}
 
-	if kogitoResources.BuildConfigS2IStatus.IsNew {
-		log.Infof("Buildconfigs are created, triggering build %s", kogitoResources.BuildConfigS2I.Name)
-		if _, err := openshift.BuildConfigC(r.client).TriggerBuild(kogitoResources.BuildConfigS2I, instance.Name); err != nil {
-			return reconcile.Result{}, err
+	deployedRes, err := kogitores.GetDeployedResources(instance, r.client)
```

Comment:
See if the new reader library can help simplify this function... usage example at https://github.com/kiegroup/kie-cloud-operator/pull/266/files#diff-9ecb910ca7304714ab23ba15d8cf55a5R791-R802

Reply:
Thanks @bmozaffa . I have modified function [GetDeployedResources](https://github.com/kiegroup/kogito-cloud-operator/blob/2db44442272539522b87d6c524e49ac09bea22b5/pkg/controller/kogitoapp/resource/deployed_resources.go#L35) to use the reader library.